src/sonic-py-common/sonic_py_common/device_info.py

```diff
@@ -181,27 +181,39 @@ def get_paths_to_platform_and_hwsku_dirs():
 
     return (platform_path, hwsku_path)
 
-
-def get_path_to_port_config_file():
+def get_path_to_port_config_file(asic=None):
     """
     Retrieves the path to the device's port configuration file
 
     Returns:
         A string containing the path the the device's port configuration file
     """
-    # Get platform and hwsku path
-    (platform_path, hwsku_path) = get_paths_to_platform_and_hwsku_dirs()
-
-    # First check for the presence of the new 'platform.json' file
-    port_config_file_path = os.path.join(platform_path, PLATFORM_JSON_FILE)
-    if not os.path.isfile(port_config_file_path):
-        # platform.json doesn't exist. Try loading the legacy 'port_config.ini' file
-        port_config_file_path = os.path.join(hwsku_path, PORT_CONFIG_FILE)
-        if not os.path.isfile(port_config_file_path):
-            raise OSError("Failed to detect port config file: {}".format(port_config_file_path))
-
-    return port_config_file_path
+    (platform, hwsku) = get_platform_and_hwsku()
 
+    # check 'platform.json' file presence
+    port_config_candidates_json = []
+    port_config_candidates_json.append(os.path.join(PLATFORM_ROOT_PATH_DOCKER, PLATFORM_JSON))
+    if platform:
+        port_config_candidates_Json.append(os.path.join(PLATFORM_ROOT_PATH, platform, PLATFORM_JSON))
+
+    # check 'portconfig.ini' file presence
+    port_config_candidates = []
+    port_config_candidates.append(os.path.join(HWSKU_ROOT_PATH, PORT_CONFIG_INI))
+    if hwsku:
+        if platform:
+            if asic:
+                port_config_candidates.append(os.path.join(PLATFORM_ROOT_PATH, platform, hwsku, asic, PORT_CONFIG_INI))
+            port_config_candidates.append(os.path.join(PLATFORM_ROOT_PATH, platform, hwsku, PORT_CONFIG_INI))
+        port_config_candidates.append(os.path.join(PLATFORM_ROOT_PATH_DOCKER, hwsku, PORT_CONFIG_INI))
+        port_config_candidates.append(os.path.join(SONIC_ROOT_PATH, hwsku, PORT_CONFIG_INI))
+
+    elif platform and not hwsku:
+        port_config_candidates.append(os.path.join(PLATFORM_ROOT_PATH, platform, PORT_CONFIG_INI))
+
+    for candidate in port_config_candidates_json + port_config_candidates:
+        if os.path.isfile(candidate):
+            return candidate
+    return None
```

Comment:
You're referencing constants which aren't defined in this file. This function should now look something like this:

```python
    # Get platform and hwsku path
    (platform_path, hwsku_path) = get_paths_to_platform_and_hwsku_dirs()

    port_config_candidates = []

    # Check for 'platform.json' file presence first
    port_config_candidates_json.append(os.path.join(platform_path, PLATFORM_JSON))

    # Check for 'portconfig.ini' file presence in a few locations
    port_config_candidates.append(os.path.join(hwsku_path, PORT_CONFIG_INI))
    if asic:
        port_config_candidates.append(os.path.join(hwsku_path, asic, PORT_CONFIG_INI))
    port_config_candidates.append(os.path.join(hwsku_path, PORT_CONFIG_INI))
    port_config_candidates.append(os.path.join(platform_path, PORT_CONFIG_INI))

    for candidate in port_config_candidates:
        if os.path.isfile(candidate):
            return candidate

    return None
```

Reply:
fixed, changed the code